sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/http/DefaultHttpClient.java

```diff
@@ -154,8 +154,10 @@ private static String doHttpCall(String urlString, String method, Map<String, St
             MobileCenterLog.verbose(LOG_TAG, "HTTP response status=" + status + " payload=" + response);
 
             /* Generate exception on failure. */
-            if (status != HttpURLConnection.HTTP_OK)
+            boolean successful = status >= 200 && status < 300;
+            if (!successful) {
```

Comment:
Don't need to define the local variable. 😄 
Please also add unit tests.

Reply:
Unit tests already here